File: src/main/kotlin/top_75/_643_maximum_average_subarray_i.java
package top_75;

public class _643_maximum_average_subarray_i {
    public double findMaxAverage(int[] nums, int k) {
        double max;
        var sum = 0;
        for (int i = 0; i < k; i++) {
            sum += nums[i];
        }
        max = (double) sum / k;
        for (int i = 0; i < nums.length - k; i++) {
            sum = sum - nums[i] + nums[i + k];
            max = Math.max(max, (double) sum / k);
        }
        return max;
    }

    public static void main(String[] args) {
        var s = new _643_maximum_average_subarray_i();
//        System.out.println(s.findMaxAverage(new int[]{1, 12, -5, -6, 50, 3}, 4));
        System.out.println(s.findMaxAverage(new int[]{0,4,0,3,2}, 1));
    }
}

File: src/main/kotlin/topic/slidingwindow/easy/_643_maximum_average_subarray_i.kt
package topic.slidingwindow.easy

import kotlin.math.max

class _643_maximum_average_subarray_i {
    fun findMaxAverage(nums: IntArray, k: Int): Double {
        var max = nums.sliceArray(0..< k).sum()
        var subsum = max
        for (i in 0 .. nums.lastIndex - k) {
            subsum = subsum - nums[i] + nums[i + k]
            max = max(max, subsum)
        }
        return max.toDouble() / k.toDouble()
    }
}

fun main() {
    with(_643_maximum_average_subarray_i()) {
        println(findMaxAverage(intArrayOf(1,12,-5,-6,50,3), 4))
        println(findMaxAverage(intArrayOf(5), 1))
    }
}
